src/Prolens.hs

```diff
@@ -214,9 +384,119 @@ infixr 4 %~
 (%~) = over
 {-# INLINE (%~) #-}
 
--- Prisms
+{- $prisms
+Prisms work with sum types.
+
+== Example
+
+Let's say we have the user data type in our system:
+
+>>> :{
+data Address = Address
+    { addressCountry :: String
+    , addressCity    :: String
+    } deriving (Show)
+:}
+
+>>> :{
+data Payload
+    = NamePayload String
+    | IdPayload Int
+    | AddressPayload Address
+    deriving (Show)
+:}
+
+To create the prism for each constructor we can use 'prism'' function and
+manually writing getter and setter function:
+
+_NOTE:_ The naming convention for prisms is the following:
+
+@
+_ConstructorName
+@
+
+>>> :{
+_NamePayload :: Prism' Payload String
+_NamePayload = prism' setter getter
+  where
+    getter :: Payload -> Maybe String
+    getter p = case p of
+        NamePayload name -> Just name
+        _otherPayload -> Nothing
+    setter :: String -> Payload
+    setter = NamePayload
```

Comment:
I think `setter` and `getter` in `Prism` should be named `construct` (or `create`) and `match`. What do you think?